packages/core/src/actions/assign.ts

```diff
@@ -177,5 +177,9 @@ export function assign<
 
   assign.resolve = resolveAssign;
 
+  assign.toJSON = () => ({
+    ...assign
+  });
```

Comment:
why do we need all of those toJSONs?

Reply:
So that these are serializable to JSON when calling `machine.definition` or `machine.toJSON()`

Reply:
But whatever you get back from this... is not JSON, it's far from it. I truly don't understand what purpose does it serve.

Reply:
This is necessary for making the tests in `json.test.ts` pass